packages/framework/attributor/src/runtimeAttributorDataStoreChannel.ts

```diff
@@ -0,0 +1,286 @@
+	 */
+	public process(
+		message: ISequencedDocumentMessage,
+		local: boolean,
+		localOpMetadata: unknown,
+	): void {
+		throw new Error("Attributor should not receive messages yet");
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.processSignal}
+	 */
+	public processSignal(message: IInboundSignalMessage, local: boolean): void {
+		throw new Error("Attributor should not receive signals");
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.summarize}
+	 */
+	public async summarize(
+		fullTree?: boolean,
+		trackState?: boolean,
+		telemetryContext?: ITelemetryContext,
+	): Promise<ISummaryTreeWithStats> {
+		return this.runtimeAttributor.summarizeOpAttributor();
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.getGCData}
+	 */
+	public async getGCData(fullGC?: boolean): Promise<IGarbageCollectionData> {
+		// Nothing to be GCed from the attributor.
+		const garbageCollectionData: IGarbageCollectionData = { gcNodes: {} };
+		return garbageCollectionData;
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.updateUsedRoutes}
+	 */
+	public updateUsedRoutes(usedRoutes: string[]): void {
+		return;
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.setConnectionState}
+	 */
+	public setConnectionState(connected: boolean, clientId?: string): void {
+		raiseConnectedEvent(this.logger, this, connected, clientId);
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.reSubmit}
+	 */
+	public reSubmit(type: string, content: unknown, localOpMetadata: unknown): void {
+		// Should not resubmit anything from the attributor as the attributor does not send ops yet.
+		throw new Error("Should not resubmit anything from the attributor");
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.applyStashedOp}
+	 */
+	public async applyStashedOp(content: unknown): Promise<unknown> {
+		// Should not apply stashed ops to the attributor as the attributor does not send ops yet.
+		throw new Error("Should not apply stashed ops to the attributor");
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.rollback}
+	 */
+	public rollback?(type: string, content: unknown, localOpMetadata: unknown): void {
+		// Should not rollback anything from the attributor as it does not send ops yet.
+		throw new Error("Should not rollback anything from the attributor");
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.entryPoint}
+	 */
+	public readonly entryPoint: IFluidHandleInternal<FluidObject>;
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.request}
+	 */
+	public async request(request: IRequest): Promise<IResponse> {
+		// Should not request anything from the attributor as the attributor does not have any channels further.
+		throw new Error("Should not request anything from the attributor");
+	}
+
+	public async waitAttached(): Promise<void> {
+		return this.deferredAttached.promise;
+	}
+
+	/**
+	 * {@inheritdoc IFluidDataStoreChannel.setAttachState}
+	 */
+	public setAttachState(attachState: AttachState.Attaching | AttachState.Attached): void {
```

Comment:
I wish you didn't have to write this code here... I know this area has been bug-prone before, and I'm not sure anyone will know to come fix any issues here if they are discovered in the similar code in the main datastore runtime implementation.

No suggestion here, just wanting to point it out.